File: backend/streamcraft/domain/transcription/ports/subtitle_parser.py
"""Subtitle parser port."""

from abc import ABC, abstractmethod
from pathlib import Path

from streamcraft.domain.shared.result import Result
from streamcraft.domain.transcription.entities.transcript import Transcript
from streamcraft.domain.transcription.errors.transcription_errors import InvalidSubtitleFormatError


class SubtitleParser(ABC):
    """Port for parsing subtitle files."""

    @abstractmethod
    def parse(self, subtitle_path: Path) -> Result[Transcript, InvalidSubtitleFormatError]:
        """Parse subtitle file to transcript."""
        ...

File: frontend/src/domain/transcription/ports/subtitle-parser.ts
/**
 * Subtitle parser port.
 */

import { Result } from '../../shared/result';
import { Transcript } from '../entities/transcript.entity';
import { InvalidSubtitleFormatError } from '../errors/transcription-errors';

export interface SubtitleParser {
    parse(subtitlePath: string): Promise<Result<Transcript, InvalidSubtitleFormatError>>;
}
